components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -409,26 +424,42 @@ public void deleteBucket(Bucket bucket) {
 
   @Override
   @Guarded(by = OPEN)
-  public BlobRef createBlob(final InputStream inputStream, Map<String, String> headers) {
+  public AssetBlob createBlob(final String blobNameHint,
```

Comment:
The significance of 'blobNameHint' (it's used for content validation) really needs a mention in the Javadoc, since it could theoretically cause this method to throw an exception.